docs/reference/configuration/single-sign-on/keycloak-customization-guide.md

```diff
@@ -0,0 +1,159 @@
+
+**Use for**: Managing Keycloak resources declaratively after deployment
+
+**Examples**:
+- Creating clients for applications
+- Managing groups and role mappings
+- Configuring identity providers
+- Automating user provisioning
+
+**Key Point**: Manages running Keycloak resources without redeployment.
+
+### Deployment Flow:
+1. **Helm values** configure Keycloak at deployment time
+2. **Identity config image** syncs custom files via init container
+3. **OpenTofu** manages resources in the running instance
+
+**Key Points**:
+- Helm values: Applied at deployment, no image rebuild needed
+- Identity config: Requires image rebuild, applied at pod startup
+- OpenTofu: Applied to running instance, no redeployment needed
+
+## Choosing the Right Approach
+
+### Choose Helm Chart Values When:
+- You need to adjust operational settings (resources, replicas, timeouts)
+- You want to enable/disable built-in authentication methods
+- You need simple branding changes (logo, favicon, background)
+- Changes are environment-specific
+- You don't want to rebuild images
+
+### Choose Identity Config Image When:
+- You need to add custom Java plugins or libraries
+- You want extensive theme customization (HTML/CSS/JS changes)
+- You need to modify the default realm structure
+- You're adding custom PKI certificates
+- You're implementing custom Keycloak SPIs
+
+### Choose OpenTofu/IaC When:
+- You want to manage Keycloak resources as code
+- You need to automate client creation for applications
+- You want GitOps workflows for identity management
+- You need to manage groups, users, and roles declaratively
+- Changes should be applied without redeployment
+
+## Common Use Cases
+
+### Simple Branding Update
+**Approach**: Helm Chart Values
+**Documentation**: [Branding Customizations](https://uds.defenseunicorns.com/reference/uds-core/idam/customization/#branding-customizations)
+
+Use ConfigMaps to override logo, favicon, and background images without rebuilding anything.
+
+### Adding a Custom Plugin
+**Approach**: Identity Config Image
+**Documentation**: [Add Additional JARs](https://uds.defenseunicorns.com/reference/uds-core/idam/customization/#add-additional-jars)
+
+1. Add JAR to `src/extra-jars/` in uds-identity-config
+2. Build custom image: `uds run build-and-publish`
+3. Override `configImage` in UDS Core bundle
+
+### Creating Application Clients
+**Approach**: OpenTofu/IaC
+**Documentation**: [OpenTofu Configuration](https://uds.defenseunicorns.com/reference/uds-core/idam/customization/#opentofu-keycloak-client-configuration)
+
+Use the Keycloak Terraform provider to manage clients declaratively.
+
+### Custom Theme Development
+**Approach**: Identity Config Image
+**Documentation**: [Customizing Theme](https://uds.defenseunicorns.com/reference/uds-core/idam/customization/#customizing-theme)
+
+1. Modify files in `src/theme/` in uds-identity-config
+2. Test locally: `uds run dev-theme`
+3. Build and publish custom image
```

Comment:
This section feels a bit repetitive with the quick reference imo. If we do keep it, I'd be curious if we think these are really the most common use-cases?
- branding update: seems reasonable as a common use-case
- custom plugin: don't think many people are doing this
- app clients: mentioned in another comment but I think this is something we would generally encourage package CR use for
- custom theme: I think kind of rare?